### PR TITLE
Add Margin Ranking Loss 

### LINKED ISSUE #531: [Feature] Need Margin Ranking Loss
I am currently working on implementing some recommendation system algorithms using MLX. During my work, I noticed that it does not include the [`MarginRankingLoss`](https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html) loss function like PyTorch, which is often critical for ranking tasks in recommendation systems.

Although I can implement this myself in my code, but maybe adding this loss would benefit other users who are working on recommendation systems and related areas?

### PR DESCRIPTION
**Pull request by AtomicVar**
## Proposed changes

This PR adds the [Margin Ranking Loss](https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html), which is commonly used for recommendation systems tasks.

The interface conforms to PyTorch, and the test ground-truth values are generated by PyTorch.

Closes #531.

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Review by Stealeristaken — APPROVED** [review]
👍🏾

**Review by angeloskath — CHANGES_REQUESTED** [review]
Looks good to me, thanks!

I left two comments. After those I think we can merge this.

**Comment by AtomicVar** [comment]
@angeloskath Thanks for your reviews. I've fixed the problems. Please let me know if you have any further feedback.

**Review by angeloskath — APPROVED** [review]

File: python/mlx/nn/losses.py [hunk 1]
```diff
@@ -532,3 +532,58 @@ def cosine_similarity_loss(
...
 
     return _reduce(loss, reduction)
+
+
+def margin_ranking_loss(
+    inputs1: mx.array,
+    inputs2: mx.array,
+    targets: mx.array,
+    margin: float = 0.0,
+    reduction: Reduction = "mean",
```
Comment (angeloskath) [inline python/mlx/nn/losses.py:542]: Oh sorry for just seeing that. All our losses have `none` as the default reduction. Could you maybe change that too.

File: python/mlx/nn/losses.py [hunk 2]
```diff
@@ -532,3 +532,58 @@ def cosine_similarity_loss(
...
+
+    Args:
+        inputs1 (array): Scores for the first input.
+        inputs2 (array): Scores for the second input.
+        targets (array): Labels indicating whether samples in ``inputs1`` should be ranked higher
+            than samples in ``inputs2``. Values should be 1 or -1.
+        margin (float, optional): The margin by which the scores should be separated.
+            Default: ``0.0``.
+        reduction (str, optional): Specifies the reduction to apply to the output:
+            ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'mean'``.
```
Comment (angeloskath) [inline python/mlx/nn/losses.py:564]: And of course in the docs as well.

**Review by angeloskath — CHANGES_REQUESTED** [review]
Sorry for just seeing that. After this tiny change I 'll do a quick merge.

File: python/mlx/nn/losses.py (hunk 1 above)
Comment (AtomicVar) [reply python/mlx/nn/losses.py]: OK, I changed it!

**Review by angeloskath — APPROVED** [review]
